### ISSUE TITLE
UI: As a future participant after the event, I will automatically get emails notifying me of subsequent events.

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/chapter#275]
Depends on #107

**Comment by allella** [comment]
The text on this [comment on #276](https://github.com/freeCodeCamp/chapter/issues/276#issuecomment-596913322) can likely be tweaked to cover upcoming _event_ notifications.

As an rough example,

```
Upcoming Events for {chapters.name}

When: {events.start_at} to {events.end_at}
Where: {venues.name}

All Upcoming Event Details: [URL to the event page, like https://{instance domain name}/chapters/:chapterId]

- Stop receiving upcoming event notifications for {chapters.name}: [URL to unsubscribe, like https://{instance domain name}/upcoming/unsubscribe/:user_id/:chapter_id/{a unique hash value to prevent 3rd party tampering}]
- More about {chapters.name} or to unfollow this chapter: [URL to chapter page, like https://{instance domain name}/chapters/:chapterId]

----------------------------
You are receiving this email because you previously followed {chapters.name}.

See the options above to change your notifications.
```

**Comment by allella** [comment]
The assumption is this would be for a single upcoming event, but it could be modified to show multiple upcoming events, or a subset of upcoming events and then link to the main events page for the specific _chapter_.

**Comment by allella** [comment]
I wanted to add the notes below since his email will likely include some options to allow _members_ to unfollow or unsubscribe a type of notification or the entire _chapter_.

The previous conversations about following / unfollowing a Chapter and unsubscribing from emails should provide good context on the type of links and privacy settings we imagined to make for easy opt-in and opt-out.

That conversation also goes into interfaces on the website and name of buttons, type of buttons and such.
For the emails, we also touched slightly on the security aspects of allowing _members_ to unsubscribe or opt-out without needing to login. For that, something like a secure / unique hash was discussed. Whether that anonymous / hash ideas is the correct approach, or easy to do is up for debate, but I thought to mention it.

The core of those email and UI opt-in / out conversations were on #107 and I tried to summarize the conversation at a few points, like:

https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-596905797
https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-562374895
https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-562432174

**Comment by bappyasif** [comment]
@allella hello again, how can i help with this issue as well? looking forward to hear from you, thanks.

**Comment by allella** [comment]
@bappyasif the comment above https://github.com/freeCodeCamp/chapter/issues/275#issuecomment-868740023 is an example of an email template that a _user_ would receive to notify them of upcoming events.

You're welcome to compare that template to similar emails you've received from other event platforms and suggest changes.

**Comment by bappyasif** [comment]
@allella are we thinking about adding some sort of maps api to go along with email as well?! that's something i can think of right now!!

**Comment by allella** [comment]
Thanks @bappyasif.  I would say a maps API is not something we've imagined for the MVP.

If you create a new issue and share links and suggests about how we might implement that feature after the MVP build, then we can add it to the [Roadmap](https://github.com/freeCodeCamp/chapter/issues?q=label%3ARoadmap).

A good first task for this issue might be to check your inbox for similar 'Upcoming Events" emails you've received in the past.  If you find any fields or links which are common, then they may be worth additional discussion.  For example, I posted [a sample above](https://github.com/freeCodeCamp/chapter/issues/275#issuecomment-868740023) based on looking at emails I'd received from a couple different event services.

We are trying to keep things simple for the MVP, but if they are simple improvements, then those would fit the intention of this issue.

**Comment by bappyasif** [comment]
@allella okay, thanks, i'll try to do some research on this and see if i can find similar fit for this probable roadmap issue

in plain words what i'm thinking about this, is:
>> to have it within what we already have within your 'sample email' where at element
>> that when user clicks it it will open their preferred 'maps service' from their device, which in turn will help them show direction and all that bits (provided maps api's allows that of course)

i also agree with you in keeping it simple and not adding anything extravagant such as maps visibility on DOM or email's real estate

i'll do my due diligence regarding this and see what i can find and see if i can create an 'issue' that might be useful for current or roadmap phase

**Comment by Sboonny** [comment]
I am wondering if there is someone working on this, before I start working on it?

**Comment by allella** [comment]
No. There are suggestions above for the wording, but otherwise I'm not aware of anybody working on it.  Thanks
